src/vbk/util_service/util_service_impl.cpp

```diff
@@ -160,10 +163,12 @@ bool UtilServiceImpl::checkCoinbaseTxWithPopRewards(const CTransaction& tx, cons
             strprintf("checkCoinbaseTxWithPopRewards(): coinbase has incorrect size of pop vouts (vouts_size=%d vs pop_vouts=%d)", tx.vout.size(), rewards.size()), "bad-pop-vouts-size");
     }
 
+    int halvings = (pindexPrev.nHeight + 1) / consensusParams.nSubsidyHalvingInterval;
+
     CAmount nTotalPopReward = 0;
     for (const auto& out : tx.vout) {
         auto it = rewards.find(out.scriptPubKey);
-        if (it != rewards.end() && it->second == out.nValue) {
+        if (it != rewards.end() && (it->second >> halvings) == out.nValue) {
             nTotalPopReward += it->second; // Pop reward
```

Comment:
you forgot to do `>> halvings` here.

Please, refactor `getPopRewards` to include this shift, so that you do not need to remember to include halvings in "client" things